source/common/http/codec_client.cc

```diff
@@ -22,7 +23,10 @@ CodecClient::CodecClient(Type type, Network::ClientConnectionPtr&& connection,
 
   ENVOY_CONN_LOG(debug, "connecting", *connection_);
   connection_->connect();
-
+  if (host != nullptr && host_->cluster().idleTimeout().valid()) {
+    idle_timer_ = dispatcher_.createTimer([this]() -> void { onIdleTimeout(); });
+    idle_timeout_ = host_->cluster().idleTimeout().value();
+  }
```

Comment:
Call enableIdleTimer() here? Or is the CodecClient always created with an active stream?

Reply:
yes. the CodecClient always created with an active stream